sdk/python/kfp/dsl/_pipeline_param.py

```diff
@@ -12,14 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import re
-from collections import namedtuple
-from typing import List, Dict, Union
-
-# TODO: Move this to a separate class
-# For now, this identifies a condition with only "==" operator supported.
-ConditionOperator = namedtuple('ConditionOperator',
-                               'operator operand1 operand2')
-PipelineParamTuple = namedtuple('PipelineParamTuple', 'name op pattern')
+import attr
+from typing import List, Dict, Union, Any
+import warnings
+
+@attr.s
+class ConditionOperator:
+  operator = attr.ib(type=str)
+  operand1 = attr.ib(type=Union['PipelineParam', Any])
+  operand2 = attr.ib(type=Union['PipelineParam', Any])
+
+  def __bool__(self) -> bool:
+    msg = ' '.join([
+      "Treating comparison of PipelineParam as bool,",
+      "but its value is not known until the pipeline is run.",
+      "If operator combinator was meant, use & or | instead."
+    ])
+    warnings.warn(msg)
+    return super().__bool__()
```

Comment:
Looks like you didn't assign a parent class to `ConditionOperator`?

Reply:
It's inherited from `object`.

Reply:
The build is failing the these errors, so probably the object didn't inherent correctly
```
Traceback (most recent call last):
  File "/home/prow/go/src/github.com/kubeflow/pipelines/sdk/python/kfp/compiler/v2_compatible_compiler_test.py", line 93, in test_two_step_pipeline
    self._assert_compiled_pipeline_equals_golden(
  File "/home/prow/go/src/github.com/kubeflow/pipelines/sdk/python/kfp/compiler/v2_compatible_compiler_test.py", line 59, in _assert_compiled_pipeline_equals_golden
skipped 9 lines unfold_more
  File "/home/prow/go/src/github.com/kubeflow/pipelines/sdk/python/kfp/dsl/_pipeline_param.py", line 32, in __bool__
    return super().__bool__()
AttributeError: 'super' object has no attribute '__bool__
```

Reply:
Looks like you're right, so it's not inheritance, but some magic instead. I'll just replace it with `True` then. Good catch!